consensus/src/chained_bft/liveness/leader_reputation_test.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::chained_bft::liveness::leader_reputation::{ReputationHeuristic, SimpleHeuristic};
+use crate::chained_bft::{
+    liveness::{
+        leader_reputation::{LeaderReputation, MetadataBackend},
+        proposer_election::ProposerElection,
+    },
+    test_utils::TestPayload,
+};
+use consensus_types::{
+    block::{block_test_utils::certificate_for_genesis, Block},
+    common::Author,
+};
+use libra_crypto::HashValue;
+use libra_types::{block_metadata::BlockMetadata, validator_signer::ValidatorSigner};
+use rand::distributions::{Distribution, WeightedIndex};
+use std::{collections::BTreeMap, sync::Arc};
+
+struct MockHistory {
+    data: Vec<BlockMetadata>,
+}
+
+impl MockHistory {
+    fn new(data: Vec<BlockMetadata>) -> Self {
+        Self { data }
+    }
+}
+
+impl MetadataBackend for MockHistory {
+    fn get_meta_data(&self, window_size: u64) -> &[BlockMetadata] {
+        let start = if self.data.len() > window_size as usize {
+            self.data.len() - window_size as usize
+        } else {
+            0
+        };
+        &self.data[start..]
+    }
+}
+
+fn create_block(proposer: Author, voters: Vec<&ValidatorSigner>) -> BlockMetadata {
+    let mut votes = BTreeMap::new();
+    for v in voters {
+        votes.insert(v.author(), v.sign_message(HashValue::zero()));
+    }
+    BlockMetadata::new(HashValue::zero(), 0, votes, proposer)
+}
+
+#[test]
```

Comment:
It would probably be good to test the situations where the committed history has a higher round than considered and ensure they agree on the proposer.

Reply:
imo that should be a test on HistoryBackend implementation which we don't have yet. It's only about returning the correct window given a target_round.

Reply:
Got it, MetadataBackend::get_meta_data() with different rounds and making sure they are the same.